environs/sync/sync.go

```diff
@@ -437,3 +394,109 @@ func (u StorageToolsUploader) UploadTools(toolsDir, stream string, tools *coreto
 	}
 	return nil
 }
+
+func UploadTestTools(
```

Comment:
Do we want this here? At the very least it needs a comment. Maybe we have a `environs/sync/testing` package?